src/tea_json.res

```diff
@@ -229,26 +229,20 @@ module Decoder = {
     },
   )
 
-
-
-
-
   let map2 = (mapper, Decoder(decoder1), Decoder(decoder2)) => Decoder(
     value => {
-      let error = x =>
-        switch x {
-        | Ok(_) => None
-        | Error(e) => Some(e)
-        }
-
-      let error_of_first = (fst, x) =>
+      switch (decoder1(value), decoder2(value)) {
+      | (Ok(v1), Ok(v2)) => Ok(mapper(v1, v2))
+      | (e1, e2) => let error_of_first = (fst, x) =>
```

Comment:
What I meant here was to integrate the error_of_first switch statement into the switch statement above it.